src/app/clusters/time-synchronization-server/time-synchronization-server.cpp

```diff
@@ -647,7 +648,9 @@ CHIP_ERROR TimeSynchronizationServer::LoadTimeZone()
 CHIP_ERROR TimeSynchronizationServer::ClearTimeZone()
 {
     InitTimeZone();
-    return mTimeSyncDataProvider.StoreTimeZone(GetTimeZone());
+    ReturnErrorOnFailure(mTimeSyncDataProvider.StoreTimeZone(GetTimeZone()));
+    emitTimeZoneStatusEvent(GetDelegate()->GetEndpoint());
```

Comment:
I think the only place this function gets called right now is on Init() if the time zone fails to load. Is the intent to have that emit an event there? That seems odd. If that IS the intent, would it make more sense to move that up into the init?

This is a public method - can the app call this? Under what circumstances? If the intent is to allow this to be called by the application code, then it should probably push this though the SetTimeZone function so it can check if the current time zone is set.

Reply:
Agree that emitting event during init is odd. Will remove it. 
My intention was that apps could decide to clear time zone for whatever reason. SetTimeZone can only be helpful if the app has the means to get a valid time zone setting without using timesync.